File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.py
class Solution:
    def findFinalValue (self, nums : List [int], original : int) -> int:
        bits = 0
        for num in nums:
            if num % original != 0:
                continue
            n = num // original
            if n & (n - 1) == 0:
                bits |= n
        d = bits + 1
        return original * (d & -d)

File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.java
class Solution {
    public int findFinalValue (int [] nums, int original) {
        int bits = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums [i] % original != 0) continue;
            nums [i] = nums [i] / original;
            if ((nums [i] & (nums [i] - 1)) == 0)
                bits |= nums[i];
        }
        int d = bits + 1;
        return original * (d & -d);
    }
}